src/freenet/node/PeerLocation.java

```diff
@@ -56,43 +60,43 @@ public synchronized int getDegree() {
 	}
 
 	boolean updateLocation(double newLoc, double[] newLocs) {
-		if(!Location.isValid(newLoc)) {
+		if (!Location.isValid(newLoc)) {
 			Logger.error(this, "Invalid location update for " + this+ " ("+newLoc+')', new Exception("error"));
 			// Ignore it
 			return false;
 		}
 
-		for(double currentLoc : newLocs) {
-			if(!Location.isValid(currentLoc)) {
-				Logger.error(this, "Invalid location update for " + this + " ("+currentLoc+')', new Exception("error"));
+		final double[] newPeersLocation = new double[newLocs.length];
+		for (int i = 0; i < newLocs.length; i++) {
+			final double loc = newLocs[i];
+			if (!Location.isValid(loc)) {
+				Logger.error(this, "Invalid location update for " + this + " (" + loc + ")", new Exception("error"));
 				// Ignore it
 				return false;
```

Comment:
I must note that there is a slight change in behaviour: we may not always return the exact same locations as the previous code did, as we traverse them in different order. The tests however ensure that we always return locations that have the exact same distance as with the previous code (namely the minimum distance).